test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -825,6 +825,29 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
         $result.Output.RelationLink["last"] | Should BeExactly "http://localhost:8080/PowerShell?test=linkheader&maxlinks=3&linknumber=3"
     }
 
+    It "Invoke-WebRequest can retry the specified number of times" {
+        try {
+            ## Setup test hook to force 2 retries.
+            [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook("FailWebRequestCount", 2)
+
+            $outputFilePath = Join-Path $TestDrive "verboseoutput.txt"
+
+            $uri = Get-WebListenerUrl -Test 'Get'
+            $command = "Invoke-WebRequest -Uri '$uri' -Verbose -RetryCount 2 -RetryIntervalSec 1 4>$outputFilePath"
+            $result = ExecuteWebCommand -command $command
+
+            ## The webrequest call succeeds after 2 retries.
+            $result.output.StatusCode | Should Be "200"
+
+            # Expect 2 verbose messages as we are retrying twice.
+            (Select-String -Path $outputFilePath -Pattern 'Retrying after interval').Count | Should Be 2
```

Comment:
This won't work once we start localizing the error messages

Reply:
I don't really like this test.  I'd rather have a real failure emulated by `WebListener`. Add a new endpoint which take a session ID (new GUID), a status code, and a fail count. Until the initially provided fail count is reached when using that session ID it produces a response with the provided status code After fail count is reached it returns a normal 200 response. The response body should include the request count for the given session ID. This way we can test how it reacts with various responses and differing fail counts provided by actual endpoint errors instead of relying on fake logic to enter the code path and without relying on verbose messages.